engine/runtime/src/test/scala/org/enso/interpreter/test/instrument/RuntimeComponentsTest.scala

```diff
@@ -244,12 +245,320 @@ class RuntimeComponentsTest
+        |""".stripMargin.linesIterator.mkString("\n")
+    val contents = metadata.appendToCode(code)
+    val mainFile = context.writeMain(contents)
+
+    // create context
+    context.send(Api.Request(requestId, Api.CreateContextRequest(contextId)))
+    context.receive shouldEqual Some(
+      Api.Response(requestId, Api.CreateContextResponse(contextId))
+    )
+
+    // open file
+    context.send(
+      Api.Request(Api.OpenFileNotification(mainFile, contents))
+    )
+    context.receiveOne shouldEqual None
+
+    // push main
+    context.send(
+      Api.Request(
+        requestId,
+        Api.PushContextRequest(
+          contextId,
+          Api.StackItem.ExplicitCall(
+            Api.MethodPointer(moduleName, "Main", "main"),
+            None,
+            Vector()
+          )
+        )
+      )
+    )
+    val responses =
+      context.receiveAllUntil(
+        context.executionComplete(contextId),
+        timeout = 120
+      )
+    // sanity check
+    responses should contain allOf (
+      Api.Response(requestId, Api.PushContextResponse(contextId)),
+      context.executionComplete(contextId)
+    )
+
+    // check the registered component groups
+    val components = context.languageContext.getPackageRepository.getComponents
+    val expectedComponents = Map(
+      LibraryName("Enso_Test", "Test") ->
+      context.pkg.config.componentGroups
+        .getOrElse(fail("Unexpected config value.")),
+      LibraryName("Standard", "Base") ->
+      RuntimeComponentsTest.standardBaseComponents,
+      LibraryName("Standard", "Test") -> ComponentGroups.empty,
+      LibraryName("Standard", "Database") ->
+      RuntimeComponentsTest.standardDatabaseComponents,
+      LibraryName("Standard", "Visualization") -> ComponentGroups.empty,
+      LibraryName("Standard", "Geo")           -> ComponentGroups.empty,
+      LibraryName("Standard", "Table") ->
+      RuntimeComponentsTest.standardTableComponents,
+      LibraryName("Standard", "Builtins") -> ComponentGroups.empty
+    )
+    components should contain theSameElementsAs expectedComponents
+
+    // check that component group symbols can be resolved
+    val suggestionSymbols = responses
+      .collect {
+        case Api.Response(
+              None,
+              msg: Api.SuggestionsDatabaseModuleUpdateNotification
+            ) =>
+          msg.updates.toVector.map(_.suggestion)
+      }
+      .flatten
+      .flatMap { suggestion =>
+        for {
+          selfType <- Suggestion.SelfType(suggestion)
+        } yield s"$selfType.${suggestion.name}"
+      }
+      .toSet
+
+    val componentSymbols = components
+      .flatMap { case (_, componentGroups) =>
+        val newComponents = componentGroups.newGroups.flatMap(_.exports)
+        val extendedComponents =
+          componentGroups.extendedGroups.flatMap(_.exports)
+        newComponents ++ extendedComponents
+      }
+      .map(_.name)
+
+    componentSymbols.foreach { component =>
```

Comment:
Maybe also check if these are non-empty, since if `componentSymbols` is empty this test will succeed but it seems like they may not have been loaded properly.